plugins/chat/app/controllers/api/chat_channels_status_controller.rb

```diff
@@ -2,17 +2,17 @@
 
 class Chat::Api::ChatChannelsStatusController < Chat::Api::ChatChannelsController
   def update
-    status = params.require(:status)
+    result =
+      Chat::Service::UpdateChannelStatus.call(
+        guardian: guardian,
+        channel: channel_from_params,
+        status: params.require(:status),
+      )
 
-    # we only want to use this endpoint for open/closed status changes,
-    # the others are more "special" and are handled by the archive endpoint
-    if !ChatChannel.statuses.keys.include?(status) || status == "read_only" || status == "archive"
-      raise Discourse::InvalidParameters
+    if result.success?
+      render_serialized(channel_from_params, ChatChannelSerializer, root: "channel")
+    else
+      # FIXME: implement failure handling
```

Comment:
Why not use the handler I introduced, `handle_service_result` ?

Reply:
mostly because I know @Flink is working on something too, so I didn't bother for now